src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -130,6 +226,20 @@ static int run(const char *dest, const char *dest_early, const char *dest_late)
                 return 0;
         }
 
+#if ENABLE_EFI_HIBERNATE_INFO
+        if (is_efi_boot()) {
+                r = parse_efi_hibernate_info();
+                if (r == -ENOMEM)
+                        return r;
+
+                /* Ensure that generic HibernateInfo will not survive more than one boot
+                 * to avoid spurious reuse. */
+                r = efi_set_variable(EFI_SYSTEMD_VARIABLE(HibernateInfo), NULL, 0);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to clear EFI variable HibernateInfo, ignoring: %m");
```

Comment:
I think we need to clear this if a normal boot happens, whether we would resume from it or not. Thus it can't be done in `systemd-sleep` or `systemd-hibernate-resume` 🤔